### ISSUE TITLE
About generated files

### ISSUE THREAD

**Issue by Soullivaneuh** [docker-php/docker-php#262]
Two things about generated files on 2.0:

- Re-generate all files per version is maybe quite overkill. Maybe it would be better to generate only once with compatible version check on constructor and setters/getters/hassers? BTW, maybe we should drop some version support.
- Generated file *should* not be versioned. Maybe create a script to call on the end-user composer.json to generate the file on install/update command?

**Comment by Soullivaneuh** [comment]
![image](https://user-images.githubusercontent.com/1698357/31886812-2787aeec-b7f6-11e7-9ca2-1b87dcb6f25b.png)

It's very very confusing.

Plus, this means we must change all the namespace at the end user while switching the docker api version.

BTW, does Docker API respects semantic versionning?

If yes, we should just support the latest stable API version. We can drop version support without any issue on a minor release: http://semver.org/#what-should-i-do-if-i-update-my-own-dependencies-without-changing-the-public-api

**Comment by Soullivaneuh** [comment]
BTW, the `docker-swagger.json` may be removed from the repository too.

**Comment by Soullivaneuh** [comment]
Another issue with this part: https://github.com/docker-php/docker-php/blob/cac4840d1d43966a5e0802e24469d806f18e08ac/src/Docker.php#L68-L188

All refer to `V1_33` namespace, but if I want to use docker with another version, on this code part:

```php
$containerCreateResult = $this->docker->container()->containerCreate(
    (new ContainersCreatePostBody())
        ->setImage("vendor/{$service}")
        ->setVolumes(
            (new ContainerConfigVolumes())
                ->setAdditionalProperties(['/repo' => (object) []])
        )
        ->setHostConfig(
            (new HostConfig())
                ->setBinds(["{$workingDirectory}:/repo"])
        )
        ->setWorkingDir('/repo')
        ->setCmd($command)
);
```

IDE is yelling because of expected class mismatch.

**Comment by joelwurtz** [comment]
I want to keep generated files versioned, as with the complete docker swagger spec it takes about 5 minutes to generate thoses files, also it offers more trust in the final client API.

To handle this i will then create a new repository which will only contain generated files, and will follow the same versioning than the API of docker

**Comment by joelwurtz** [comment]
See last version @Soullivaneuh this should be better, i only generated 1.25 but it gives a good example, closing this as it's not releveant anymore

**Comment by Soullivaneuh** [comment]
@joelwurtz Didn't get time to check it, but it looks indeed better for the namespaces. :+1: 

But why the package version is `4.1.25`? What v4 is meant here? Plus, it's not really semver if the generated code change for the next version remove some methods. It should be at least notified on the documentation IMO.

**Comment by joelwurtz** [comment]
it's specified on the https://github.com/docker-php/docker-php-api README, did not get the time to write docs for the 2.0 version.

**Comment by Soullivaneuh** [comment]
I just checked, it works very well with nice smooth namespaces. Good job! :+1:
